webapp/src/features/adminAddIntersection/adminAddIntersectionSlice.tsx

```diff
@@ -0,0 +1,256 @@
+import { PayloadAction, createAsyncThunk, createSlice } from '@reduxjs/toolkit'
+import { selectToken } from '../../generalSlices/userSlice'
+import EnvironmentVars from '../../EnvironmentVars'
+import apiHelper from '../../apis/api-helper'
+import { updateTableData as updateIntersectionTableData } from '../adminIntersectionTab/adminIntersectionTabSlice'
+import { RootState } from '../../store'
+import { AdminAddIntersectionForm } from './AdminAddIntersection'
+
+export type AdminIntersectionCreationInfo = {
+  organizations: string[]
+  rsus: string[]
+}
+
+export type AdminIntersectionKeyedCreationInfo = {
+  organizations: { id: number; name: string }[]
+  rsus: { id: number; name: string }[]
+}
+
+export type AdminIntersectionCreationBody = {
+  intersection_id: string
+  ref_pt: {
+    latitude: string
+    longitude: string
+  }
+  bbox?: {
+    latitude1: string
+    longitude1: string
+    latitude2: string
+    longitude2: string
+  }
+  intersection_name?: string
+  origin_ip?: string
+  organizations: string[]
+  rsus: string[]
+}
+
+const initialState = {
+  apiData: {} as AdminIntersectionKeyedCreationInfo,
+  selectedOrganizations: [] as AdminIntersectionKeyedCreationInfo['organizations'],
+  selectedRsus: [] as AdminIntersectionKeyedCreationInfo['rsus'],
+  submitAttempt: false,
+}
+
+export const updateApiJson = (apiJson: AdminIntersectionCreationInfo): AdminIntersectionKeyedCreationInfo => {
```

Comment:
Can we add some descriptions to the methods? "UpdateApiJson" is fairly ambiguous as to what the method does.

Reply:
Updated the name to "convertApiJsonToKeyedFormat"